openstates/de/legislators.py

```diff
@@ -68,64 +92,64 @@ def scrape_bio(self, term, chamber, district, name, url):
         return leg
 
     def scrape_contact_info(self, doc):
-        office_names = ['Legislative Hall Office', 'Outside Office']
-        office_types = ['capitol', 'district']
-        xpath = '//u[contains(., "Office")]/ancestor::table/tr[2]/td'
-        data = zip(doc.xpath(xpath)[::2], office_names, office_types)
-        info = {}
 
         # Email
-        xpath = '//font[contains(., "E-mail Address")]/../font[2]'
-        email = doc.xpath(xpath)[0].text_content()
-
-        # If multiple email addresses listed, only take the official
-        # noone@state.de.us address.
-        emails = re.split(r'(?:\n| or |;|\s+)', email)
-        for email in filter(None, emails):
-            if email.strip():
-                info['email'] = email.strip()
-                break
+        email = doc.xpath(".//a[contains(@href,'mailto')]")
+        email = email[0].text_content().strip()
+        if not email:
+            email = None
 
         # Offices
-        offices = []
-        for (td, name, type_) in data:
-            office = dict(name=name, type=type_, phone=None,
-                          fax=None, email=None, address=None)
-
-            chunks = td.text_content().strip().split('\n\n')
-            chunks = [s.strip() for s in chunks]
-            chunks = filter(None, chunks)
-            if len(chunks) == 1:
-                if ':' in chunks[0].splitlines()[0]:
-                    # It's just phone numbers with no actual address.
-                    numbers = [chunks[0]]
-                    office['address'] = None
-                elif chunks[0].strip().startswith("("):
-                    numbers = [chunks[0]]
-                    office['address'] = None
+
+        leg_office = dict(name="Capitol Office", type="capitol",
+                        phone=None, fax=None, email=email, address=None)
+        dist_office = dict(name="District Office", type="district",
+                        phone=None,fax=None, email=email, address=None) 
```

Comment:
I've thought this through for the guy with two e-mail addresses and am now doing something reasonable.